Doesn't work on Java9 and macOS

The following error occurred when running with GUI mode on Java9 and macOS.

```
Exception in thread "main" java.lang.NoClassDefFoundError: com/apple/eawt/AboutHandler
	at stew5.ui.swing.WindowLauncher.<init>(WindowLauncher.java:72)
	at stew5.ui.swing.WindowLauncher.main(WindowLauncher.java:887)
	at stew5.App.run(App.java:110)
	at stew5.App.main(App.java:133)
Caused by: java.lang.ClassNotFoundException: com.apple.eawt.AboutHandler
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:582)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:185)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:496)
	... 4 more
```

Environment:
```
JRE : Java(TM) SE Runtime Environment 9+181
OS : Mac OS X (osver=10.12.6)
Locale : ja_JP
```

--- a/build.gradle
+++ b/build.gradle
@@ -10,7 +10,6 @@ dependencies {
     compile 'commons-cli:commons-cli:1.4'
     compile 'jline:jline:2.14.3'
 	compile 'org.apache.commons:commons-csv:1.4'
-	compile 'com.yuvimasory:orange-extensions:1.3.0'
     testCompile 'junit:junit:4.12'
     testCompile 'org.hamcrest:hamcrest-junit:2.0.0.0'
     testCompile 'com.h2database:h2:1.4.191'

--- a/src/main/java/stew5/ui/swing/AppleMenu.java
+++ b/src/main/java/stew5/ui/swing/AppleMenu.java
@@ -1,0 +1,95 @@
+package stew5.ui.swing;
+
+import java.awt.*;
+import java.lang.reflect.*;
+import java.util.*;
+import java.util.Map.*;
+import javax.script.*;
+import javax.swing.*;
+import stew5.*;
+
+final class AppleMenu {
+
+    private static final Logger log = Logger.getLogger(Menu.class);
+    private static final ResourceManager res = ResourceManager.getInstance(Menu.class);
+
+    private AppleMenu() {
+    }
+
+    public static void customize(Menu menu,
+                                 int javaMajorVersion,
+                                 Map<String, JMenuItem> itemMap,
+                                 final AnyActionListener anyActionListener) {
+        log.info("customizing for macOS, java version=%d", javaMajorVersion);
+        ScriptEngine se = new ScriptEngineManager().getEngineByExtension("js");
+        se.put("d", new Delegate(menu, anyActionListener));
+        final String gettingAppCode;
+        if (javaMajorVersion >= 9) {
+            log.debug("OrangeMenu.applyCustomize() version 9+");
+            gettingAppCode = "java.awt.Desktop.getDesktop()";
+        } else {
+            log.debug("OrangeMenu.applyCustomize() version 8-");
+            gettingAppCode = "com.apple.eawt.Application.getApplication()";
+        }
+        final String s;
+        s = ""
+            + "var app = "
+            + gettingAppCode
+            + ";"
+            + "app.setAboutHandler(function(evt) {"
+            + "  d.showAbout(evt.source);"
+            + "});"
+            + "app.setQuitHandler(function(evt,qr) {"
+            + "  if (d.confirmQuitAndYes()) {"
+            + "    d.quit();"
+            + "    qr.performQuit();"
+            + "  } else {"
+            + "    qr.cancelQuit();"
+            + "  }"
+            + "});"
+            + "d.setupDockerIcon();"
+            + "";
+        try {
+            se.eval(s);
+        } catch (ScriptException e) {
+            throw new RuntimeException(e);
+        }
+        for (Entry<String, JMenuItem> entry : new ArrayList<>(itemMap.entrySet())) {
+            String key = entry.getKey();
+            if (key.equals(AnyActionKey.quit.toString()) || key.equals(AnyActionKey.showAbout.toString())) {
+                itemMap.remove(key);
+                JMenuItem item = entry.getValue();
+                item.getParent().remove(item);
+            }
+        }
+    }
+
+    // must declare as public to be enable to call by nashorn
+    public static final class Delegate {
+        private final Component parent;
+        private final AnyActionListener anyActionListener;
+        public Delegate(Component parent, AnyActionListener anyActionListener) {
+            this.parent = parent;
+            this.anyActionListener = anyActionListener;
+        }
+        public boolean confirmQuitAndYes() {
+            return JOptionPane.showConfirmDialog(parent,
+                                                 res.get("i.confirm-quit"),
+                                                 "",
+                                                 JOptionPane.YES_NO_OPTION) == JOptionPane.YES_OPTION;
+        }
+        public void quit() {
+            WindowLauncher.exit();
+        }
+        public void showAbout(Object srcComponent) {
+            anyActionListener.anyActionPerformed(new AnyActionEvent(srcComponent, Menu.Item.showAbout));
+        }
+        public void setupDockerIcon() throws Exception {
+            Class<?> c = Class.forName("com.apple.eawt.Application");
+            Method m = c.getDeclaredMethod("getApplication");
+            Method m2 = c.getDeclaredMethod("setDockIconImage", Image.class);
+            m2.invoke(m.invoke(null), Utilities.getImageIcon("stew.png").getImage());
+        }
+    }
+
+}

--- a/src/main/java/stew5/ui/swing/Menu.java
+++ b/src/main/java/stew5/ui/swing/Menu.java
@@ -1,6 +1,5 @@
 package stew5.ui.swing;
 
-import static stew5.ui.swing.AnyActionKey.showAbout;
 import static stew5.ui.swing.Menu.Item.*;
 import static stew5.ui.swing.Utilities.*;
 import java.awt.*;
@@ -8,11 +7,9 @@ import java.beans.*;
 import java.io.*;
 import java.util.*;
 import java.util.List;
-import java.util.Map.Entry;
+import java.util.Map.*;
 import java.util.regex.*;
 import javax.swing.*;
-import com.apple.eawt.*;
-import com.apple.eawt.AppEvent.*;
 import stew5.*;
 
 /**
@@ -309,40 +306,29 @@ final class Menu extends JMenuBar implements PropertyChangeListener {
         return !App.props.getAsBoolean("ui.suppressGenerateMnemonic") && res.getInt("auto-mnemonic") == 1;
     }
 
+    interface MenuCustomizer {
+        void customize(Map<String, JMenuItem> itemMap, AnyActionListener anyActionListener);
+    }
+
+    MenuCustomizer getMenuCustomizer(String simpleClassName) {
+        try {
+            final String fqcn = getClass().getPackage().getName() + "." + simpleClassName;
+            final Object o = DynamicLoader.loadClass(fqcn).newInstance();
+            return (MenuCustomizer)o;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
     private void customize(Map<String, JMenuItem> itemMap, final AnyActionListener anyActionListener) {
-        if (System.getProperty("os.name", "").regionMatches(true, 0, "Mac OS X", 0, 8)) {
-            class OrangeQuitHandler implements QuitHandler {
-                @Override
-                public void handleQuitRequestWith(QuitEvent evt, QuitResponse qr) {
-                    if (JOptionPane.showConfirmDialog(Menu.this.getParent(),
-                                                      res.get("i.confirm-quit"),
-                                                      "",
-                                                      JOptionPane.YES_NO_OPTION) == JOptionPane.YES_OPTION) {
-                        WindowLauncher.exit();
-                        qr.performQuit();
-                    } else {
-                        qr.cancelQuit();
-                    }
-                }
+        try {
+            final String javaVersionString = System.getProperty("java.runtime.version", "0");
+            final int javaMajorVersion = Integer.parseInt(javaVersionString.replaceFirst("^(\\d+).+?$", "$1"));
+            if (System.getProperty("os.name", "").regionMatches(true, 0, "Mac OS X", 0, 8)) {
+                AppleMenu.customize(this, javaMajorVersion, itemMap, anyActionListener);
             }
-            class OrangeAboutHandler implements AboutHandler {
-                @Override
-                public void handleAbout(AboutEvent evt) {
-                    anyActionListener.anyActionPerformed(new AnyActionEvent(evt.getSource(), showAbout));
-                }
-            }
-            Application app = Application.getApplication();
-            app.setDockIconImage(Utilities.getImageIcon("stew.png").getImage());
-            app.setAboutHandler(new OrangeAboutHandler());
-            app.setQuitHandler(new OrangeQuitHandler());
-            for (Entry<String, JMenuItem> entry : new ArrayList<>(itemMap.entrySet())) {
-                String key = entry.getKey();
-                if (key.equals(AnyActionKey.quit.toString()) || key.equals(AnyActionKey.showAbout.toString())) {
-                    itemMap.remove(key);
-                    JMenuItem item = entry.getValue();
-                    item.getParent().remove(item);
-                }
-            }
+        } catch (Throwable th) {
+            WindowOutputProcessor.showErrorDialog(this.getParent(), th);
         }
     }
 

--- a/src/main/java/stew5/ui/swing/Menu.java
+++ b/src/main/java/stew5/ui/swing/Menu.java
@@ -306,20 +306,6 @@ final class Menu extends JMenuBar implements PropertyChangeListener {
         return !App.props.getAsBoolean("ui.suppressGenerateMnemonic") && res.getInt("auto-mnemonic") == 1;
     }
 
-    interface MenuCustomizer {
-        void customize(Map<String, JMenuItem> itemMap, AnyActionListener anyActionListener);
-    }
-
-    MenuCustomizer getMenuCustomizer(String simpleClassName) {
-        try {
-            final String fqcn = getClass().getPackage().getName() + "." + simpleClassName;
-            final Object o = DynamicLoader.loadClass(fqcn).newInstance();
-            return (MenuCustomizer)o;
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
-    }
-
     private void customize(Map<String, JMenuItem> itemMap, final AnyActionListener anyActionListener) {
         try {
             final String javaVersionString = System.getProperty("java.runtime.version", "0");

